### DISCUSSION TITLE
How to access to map hints to kitty's buffer?

### DISCUSSION THREAD

**Discussion by Cruising0904** [kovidgoyal/kitty#6073] [Q&A] [answered]
I found this useful feature kitten hints. but if I make this even more beautiful I have to map it to kitty buffer on letter 
just like copy_to_buffer a <-- Can I implement this right after hint pressed? 

I looked up for this feature whole day today, couldn't find to a way to access buffer copy thing 

so far I have mapped
![image](https://user-images.githubusercontent.com/29267295/221767664-7a0cdfb2-3fd5-41a6-96a5-209309bc0ed6.png)
![image](https://user-images.githubusercontent.com/29267295/221767716-ce9b9de7-deea-4780-9679-4ad0e275c164.png)
like this. 

How am I going to make this work?

**Comment by page-down** [comment, marked as answer]
UPDATE:
Now hints kitten can copy the matches to a named buffer.

https://github.com/kovidgoyal/kitty/pull/6078

```conf
map f1 kitten hints --program @a
```

---
You need to customize the action performed by hints kitten.

https://sw.kovidgoyal.net/kitty/kittens/hints/#completely-customizing-the-matching-and-actions-of-the-kitten
https://sw.kovidgoyal.net/kitty/kittens/hints/#cmdoption-kitty-kitten-hints-customize-processing

Put the retrieved text into `boss.clipboard_buffers[buffer_name]`.
Search for `copy_to_buffer`, `paste_from_buffer` in the codebase to learn more.

The option `--program` follows an executable on the system or an explicitly stated special value, not the kitty's actions.

**Reply by Cruising0904** [reply]
Thank you ! will try it :)

**Reply by Cruising0904** [reply]
I just saw this post yesterday and I tried this (not customized script one) didn't trigger error though can not bring from named buffer. Seems like it was fixed but may I know how you did this? 
![image](https://user-images.githubusercontent.com/29267295/229964148-4cbe65b3-8cfb-4aa4-804a-f05c6dbcd02f.png)
![image](https://user-images.githubusercontent.com/29267295/229964198-79a7fa25-8f08-4718-be17-fd4d2cd16b0a.png)

I could manage to copy to buffer and paste one, couldn't paste from hint

**Reply by kovidgoyal** [reply]
You need to run kitty nightly.

**Reply by Cruising0904** [reply]
I see. Thanks! :)

**Reply by Cruising0904** [reply]
> UPDATE: Now hints kitten can copy the matches to a named buffer.
> 
> #6078
> 
> ```
> map f1 kitten hints --program @a
> ```
> 
> You need to customize the action performed by hints kitten.
> 
> https://sw.kovidgoyal.net/kitty/kittens/hints/#completely-customizing-the-matching-and-actions-of-the-kitten https://sw.kovidgoyal.net/kitty/kittens/hints/#cmdoption-kitty-kitten-hints-customize-processing
> 
> Put the retrieved text into `boss.clipboard_buffers[buffer_name]`. Search for `copy_to_buffer`, `paste_from_buffer` in the codebase to learn more.
> 
> The option `--program` follows an executable on the system or an explicitly stated special value, not the kitty's actions.

thank you so much! I really appreciate :+1:
